File: backend/models/Question.js
const mongoose = require('mongoose');

const QuestionSchema = new mongoose.Schema({
  question: {
    type: String,
    required: true,
    trim: true
  },
  optionA: {
    type: String,
    required: true,
    trim: true
  },
  optionB: {
    type: String,
    required: true,
    trim: true
  },
  optionC: {
    type: String,
    default: '',
    trim: true
  },
  optionD: {
    type: String,
    default: '',
    trim: true
  },
  options: [{
    type: String,
    trim: true
  }],
  correctAnswer: {
    type: String,
    required: true
  },
  category: {
    type: String,
    required: true,
    trim: true
  },
  difficulty: {
    type: String,
    required: true,
    enum: ['easy', 'medium', 'hard']
  },
  imageUrl: {
    type: String,
    default: null
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Question', QuestionSchema);

File: frontend/app/src/main/java/com/ilygames/quizapp/data/model/Question.kt
package com.ilygames.quizapp.data.model

import com.google.gson.annotations.SerializedName

data class Question(
    @SerializedName("_id") val id: String? = null,
    val question: String,
    val optionA: String = "",
    val optionB: String = "",
    val optionC: String = "",
    val optionD: String = "",
    val options: List<String>? = null,
    val correctAnswer: String, // "A", "B", "C", "D", "E", "F"
    val category: String = "General",
    val difficulty: String = "easy",
    val imageUrl: String? = null
) {
    fun getOptionsList(): List<String> {
        if (!options.isNullOrEmpty()) {
            return options.filter { it.isNotBlank() }
        }
        return listOf(optionA, optionB, optionC, optionD).filter { it.isNotBlank() }
    }
}
